src/main/java/com/jwt/jwt/Controller/JwtAuthenticationController.java

```diff
@@ -0,0 +1,50 @@
+package com.jwt.jwt.Controller;
+
+import com.jwt.jwt.Entity.UserData;
+import com.jwt.jwt.JwtConfig.JwtResonce;
+import com.jwt.jwt.JwtConfig.JwtTokenUtil;
+import com.jwt.jwt.JwtConfig.JwtUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("")
+public class JwtAuthenticationController {
+    @Autowired
+    private AuthenticationManager authenticationManager;
+    @Autowired
+    private JwtTokenUtil jwtTokenUtil;
+
+    @Autowired
+    private JwtUserDetailsService jwtUserDetailsService;
+
+    @PostMapping("/login")
+    public ResponseEntity<?> createAuthToken(@RequestBody UserData userData)throws Exception{
```

Comment:
If we use @RestController annotation then do we have to use ResponseEntity as the return type?
Please read about what is the difference between RestController and Controller annotation.

Reply:
ResponseEntity removed